cmd/controller-manager/app/controller-manager.go

```diff
@@ -177,6 +183,45 @@ func startControllers(opts *options.Options, stopChan <-chan struct{}) {
 	}
 }
 
+func setOptionsByFederationConfig(opts *options.Options) {
+	client := genericclient.NewForConfigOrDieWithUserAgent(opts.Config.KubeConfig, "federationconfig")
+
+	name := util.FederationConfigName
+	namespace := opts.Config.FederationNamespace
+	fedConfig := &corev1a1.FederationConfig{}
+	err := client.Get(context.Background(), fedConfig, namespace, name)
+	if err != nil {
+		glog.V(1).Infof("Cannot retrieve FederationConfig %q in namespace %q: %v. Command line options are used.", name, namespace, err)
```

Comment:
(No action required) Are you intending to remove the cli args at some point?

Reply:
Yes. It is agreement during our discuss as far as I remember. I will update when clean up options.